components/dashboard/src/settings/Teams.tsx

```diff
@@ -0,0 +1,856 @@
+
+    const [slots, setSlots] = useState<Slot[]>([]);
+    const [showPaymentUI, setShowPaymentUI] = useState<boolean>(false);
+    const [isChargebeeCustomer, setIsChargebeeCustomer] = useState<boolean>(false);
+    const [isStudent, setIsStudent] = useState<boolean>(false);
+    const [teamSubscriptions, setTeamSubscriptions] = useState<TeamSubscription[]>([]);
+
+    const [createTeamModal, setCreateTeamModal] = useState<{ types: string[], defaultCurrency: string } | undefined>(undefined);
+    const [manageTeamModal, setManageTeamModal] = useState<{ sub: TeamSubscription } | undefined>(undefined);
+    const [inviteMembersModal, setInviteMembersModal] = useState<{ sub: TeamSubscription } | undefined>(undefined);
+    const [addMembersModal, setAddMembersModal] = useState<{ sub: TeamSubscription } | undefined>(undefined);
+
+    const [pendingPlanPurchase, setPendingPlanPurchase] = useState<{ plan: Plan } | undefined>(getLocalStorageObject<{ plan: Plan }>("pendingPlanPurchase"));
+    const [pendingSlotsPurchase, setPendingSlotsPurchase] = useState<{ ts: TeamSubscription } | undefined>(getLocalStorageObject<{ ts: TeamSubscription }>("pendingSlotsPurchase"));
+
+    useEffect(() => {
+        queryState();
+    }, []);
+
+    useEffect(() => {
+        if (pendingPlanPurchase) {
+            setLocalStorageObject("pendingPlanPurchase", pendingPlanPurchase);
+        } else {
+            removeLocalStorageObject("pendingPlanPurchase");
+        }
+    }, [pendingPlanPurchase]);
+
+    useEffect(() => {
+        if (pendingSlotsPurchase) {
+            setLocalStorageObject("pendingSlotsPurchase", pendingSlotsPurchase);
+        } else {
+            removeLocalStorageObject("pendingSlotsPurchase");
+        }
+    }, [pendingSlotsPurchase]);
+
+    const queryState = async () => {
+        try {
+            const [slots, showPaymentUI, teamSubscriptions, clientRegion, isStudent] = await Promise.all([
+                getGitpodService().server.tsGetSlots(),
+                getGitpodService().server.getShowPaymentUI(),
+                getGitpodService().server.tsGet(),
+                getGitpodService().server.getClientRegion(),
+                getGitpodService().server.isStudent(),
+            ]);
+    
+            setDefaultCurrency((clientRegion && (countries as any)[clientRegion]?.currency === 'EUR') ? 'EUR' : 'USD');
+            setIsStudent(isStudent);
+    
+            setSlots(slots);
+            setShowPaymentUI(showPaymentUI);
+            setTeamSubscriptions(teamSubscriptions);
+        } catch (error) {
+            console.log(error);
+        }
+    }
+
+    useEffect(() => {
+        if (!isChargebeeCustomer) {
+            (async () => {
+                try {
+                    setIsChargebeeCustomer(await getGitpodService().server.isChargebeeCustomer());
+                } catch (error) {
+                    console.log(error);
+                }
+            })()
+        }
+        if (pendingPlanPurchase) {
```

Comment:
Suggestion: Here you could remove any pending state that has expired, e.g. like so:

```suggestion
        if (pendingPlanPurchase) {
            // Consider any pending purchase to be expired after 5 minutes
            if (pendingPlanPurchase.pendingSince && (pendingPlanPurchase.pendingSince + 1000 * 60 * 5) > Date.now()) {
                setPendingPlanPurchase(undefined);
            }
```